### PR TITLE
[DO NOT MERGE] Test NVIDIA's official CUDA packages

### LINKED ISSUE #139: CuPy: Support CUDA 12
Creating an issue so that I won't forget...
- The main discussion issue: conda-forge/cudatoolkit-feedstock#62
- @jaimergp's attempt: conda-forge/openmm-feedstock#60

Let's do this and get prepared for the upcoming GPU CI (https://github.com/conda-forge/conda-forge.github.io/issues/1272).

### PR DESCRIPTION
**Pull request by leofang**
Close #139.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
Missing `crt/host_defines.h`:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=364599&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=205
Reported in https://github.com/conda-forge/cudatoolkit-feedstock/issues/62#issuecomment-899959254.

**Comment by leofang** [comment]
Tests passed on Linux64 and Windows.

**Comment by leofang** [comment]
Darn it, I was fooled by the ongoing Azure issue. The feedstock status is not reported, so the green check only includes the linter's report...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -93,15 +96,25 @@ requirements:
...
+    #{% if cuda_major == 11 %}
+    #- cusparselt  # [linux64]
+    #{% endif %}
+    #{% if cuda_major_minor >= (10, 1) %}
+    #- libcugraph  # [linux64]
+    #{% endif %}
+    - cuda-nvcc  # for crt/host_defines.h
+    - cuda-cudart
+    - cuda-nvrtc
+    - cuda-nvprof  # for cuda_profiler_api.h
```
Comment (jakirkham) [inline recipe/meta.yaml:111]: If we have the `cuda-profiler-api` package, does that solve this need? Or are there other things that `cuda-nvprof` is still needed for?

Comment (leofang) [reply recipe/meta.yaml:111]: I think CuPy doesn't need the legacy `nvprof` (cmdline tool), just the profiler header.

**Comment by leofang** [comment]
Would be superseded by #199.

**Pull request closed without merging**